Cython/Compiler/TypeSlots.py

```diff
@@ -503,6 +503,23 @@ def spec_value(self, scope):
         return self.slot_code(scope)
 
 
+class DelSlot(SyntheticSlot):
+    # For __del__ if defined
+    def __init__(self, slot_name, user_methods, default_value, **kwargs):
+        super(DelSlot, self).__init__(slot_name, user_methods, default_value, **kwargs)
+
+    def slot_code(self, scope):
+        if not scope.lookup_here("__del__"):
+            return 0
```

Comment:
Another advantage right here: this can go away (and the whole method, actually), if the C conditions are handled in `generate()`.